acme4j-client/src/main/java/org/shredzone/acme4j/Certificate.java

```diff
@@ -131,6 +147,39 @@ public void writeCertificate(@WillNotClose Writer out) throws IOException {
         }
     }
 
+    /**
+     * Writes the certificate to the given writer. It is written in PEM format.
+     *
+     * @param out
+     *            {@link Writer} to write to. The writer is not closed after use.
+     */
+    public void writeCertificateOnly(@WillNotClose Writer out) throws IOException {
+        try {
+            AcmeUtils.writeToPem(getCertificate().getEncoded(), AcmeUtils.PemLabel.CERTIFICATE, out);
+        } catch (CertificateEncodingException ex) {
+            throw new IOException("Encoding error", ex);
+        }
+    }
+
+    /**
+     * Writes the intermediate certificates to the given writer. They are written in
+     * PEM format. The list is sorted, following certificates certify preceding ones.
+     *
+     * @param out
+     *            {@link Writer} to write to. The writer is not closed after use.
+     */
+    public void writeChain(@WillNotClose Writer out) throws IOException {
+        try {
+            for (X509Certificate cert : getCertificateChain()) {
+                if (cert != getCertificate()) {
```

Comment:
I don't feel really well to use `!=` here, even if it works. It may break silently if `getCertificate()` returns a different instance than `getCertificateChain()` some day. The first object of `getCertificateChain()` is always the certificate itself, so you can just write out starting from the second object. (The list might contain only one element, though.)